src/context.rs

```diff
@@ -26,12 +26,12 @@ use ffi::enums::{Status, Antialias, LineCap, LineJoin, FillRule};
 use ::patterns::{wrap_pattern, Pattern};
 use surface::Surface;
 
-pub struct RectangleVec {
+pub struct RectangleVec<'a> {
     ptr: *mut cairo_rectangle_list_t,
-    pub rectangles: CVec<Rectangle>,
+    pub rectangles: &'a[Rectangle], // Vec<Rectangle>,
 }
```

Comment:
>So borrow checker think that RectangleVec can't outlive Context object for that function was called.

... what magic is this? But good to know. Yeah, my intent was to have it be a refcounted object independent of Context (I checked cairo's copy_clip_rectangle_list - it just mallocs a list). But apparently that works in a different way. Sorry.